ingresspipe/models/metadata.py

```diff
@@ -0,0 +1,262 @@
+import pandas as pd
+import networkx as nx
+import json
+import re
+
+from jsonschema import Draft7Validator, exceptions, validate, ValidationError
+
+# allows specifying explicit variable types
+from typing import Any, Dict, Optional, Text, List
+
+# handle schema logic; to be refactored as SchemaExplorer matures into a package
+# as collaboration with Biothings progresses
+
+import sys
+sys.path.append("..") # Adds higher directory to python modules path.
+
+from ingresspipe.schemas.explorer import SchemaExplorer
+from ingresspipe.manifest.generator import ManifestGenerator
+from ingresspipe.schemas.generator import SchemaGenerator
+
+class MetadataModel(object):
+    """Metadata model wrapper around schema.org specification graph.
+    
+    Provides basic utilities to:
+
+    1) manipulate the metadata model;
+    2) generate metadata model views:
+        - generate manifest view of the metadata model
+        - generate validation schema view of the metadata model;
+    """
+
+    def __init__(self,
+                inputMModelLocation: str,
+                inputMModelLocationType: str,
+                ) -> None:
+
+        """Instantiates a MetadataModel object.
+
+        Args:
+            inputMModelLocation: local path, uri, synapse entity id; (e.g. gs://, syn123, /User/x/…); present location
+            inputMModelLocationType: specifier to indicate where the metadata model resource can be found; (e.g. 'local' if file/JSON-LD is on local machine)
+        """
+        # extract extension of 'inputMModelLocation'
+        # ensure that it is necessarily pointing to a '.jsonld' file
+        if inputMModelLocation.rpartition('.')[-1] == "jsonld":
+            self.inputMModelLocation = inputMModelLocation
+
+            self.sg = SchemaGenerator(inputMModelLocation)
+        else:
+            print("Can't create object of SchemaGenerator class. Please make sure the 'inputMModelLocation' argument is pointing to a JSON-LD file.")
```

Comment:
Just a note, not a blocker: we should use the Python `logging` library for console messages instead of `print`.

Reply:
Thanks! Will make sure to change it everywhere accordingly.

Reply:
Thanks, Sujay! OK to save for a separate PR, if you want to get this one wrapped up more quickly.